dts/arm/st/pinctrl_st_stm32.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2020 Linaro Limited
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef PINCTRL_ST_STM32_H_
+#define PINCTRL_ST_STM32_H_
+
+/* Create a pincfg device tree node:
+ *
+ * The node name and nodelabel will be of the form:
+ *
+ * NODE = <periph>_<signal>_p<port>_<pin>
+ *
+ * NODE: NODE {
+ *     pinmux = <STM32_PINMUX_Z(STM32_DT_PORT_<port>,pin, function)>;
+ *
+ * }
+ *
+ * Also, each macro will fill directly a knwon working pin configuration
+ * depending on the signal that is required.
+ *
+ * So for example:
+ *
+ * DT_STM32_UART_TX(b,6,usart1,AF7);
+ *
+ * Will become:
+ *
```

Comment:
It could be interresting to represent how information is encoded, something like that:
bits [15..12] port number
bits [11..8] pin number
bits [7..0] Alternate function

Reply:
Information is also provided in dts/bindings/pinctrl/st,stm32-pinctrl.yaml, so I'm not sure